CopyFiles_TPL/CopyFiles_TPL/ViewModel.cs

```diff
@@ -0,0 +1,275 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Documents;
+using System.Windows.Forms;
+using System.Windows.Threading;
+using OpenFileDialog = Microsoft.Win32.OpenFileDialog;
+using SaveFileDialog = Microsoft.Win32.SaveFileDialog;
+
+namespace CopyFiles
+{
+    class ViewModel : ViewModelBase
+    {
+        private readonly Progress<long> _progress;
+        private int progress;
+        private long _currentlyCopied;
+        private long _totalFilesSize;
+        private string _sourcePath = String.Empty;
+        private string _destinationPath = String.Empty;
+        private string _folderToCreate = String.Empty;
+        private readonly CancellationTokenSource _cancelTokenSource;
+
+
+        public ViewModel()
+        {
+            _cancelTokenSource = new CancellationTokenSource();
+            _progress = new Progress<long>();
+            _progress.ProgressChanged += (s, e) =>
+            {
+                _currentlyCopied += e;
+                Progress = (int)((double)_currentlyCopied / _totalFilesSize * 100);
+            };
+        }
+
+        #region Properties
+
+        public int Progress
+        {
+            get => progress;
+            private set
+            {
+                progress = value;
+                if (progress == 100)
+                {
+                    progress = 0;
+                    _currentlyCopied = 0;
+                }
+                OnPropertyChanged();
+            }
+        }
+
+        public string SourcePath
+        {
+            get => _sourcePath;
+            set
+            {
+                _sourcePath = value;

```

Comment:
You should compare the current value with the setting one and raise the event only if they are not same.